### PR TITLE
Feature 3 : Integrated All GET APIs with RTK-Query

### PR DESCRIPTION
**Pull request by Abhishek-Jain-1925**
Feature 3 : Integrated All GET APIs with RTK-Query

Work done are as follows - 
1. All well-defined forms as per new updates
2. Integration of All GET APIs with RTK-Query
3. Overall UI optimization
4. Removed unnecessary code snippets

### CODE REVIEW

File: src/Constants.js [hunk 1]
```diff
@@ -8,7 +8,13 @@ export const PROFILES = {
 
 // API ROUTES
 export const LOGIN_ENDPOINT = "/login";
-export const PROFILE_LIST_ENDPOINT = "/api/list_profiles";
+export const PROFILE_LIST_ENDPOINT = "/api/profiles";
+export const PROFILE_ENDPOINT = "/api/profiles/";
+export const PROJECT_ENDPOINT = "/projects";
```
Comment (Payal13) [inline src/Constants.js]: Please mention the complete endpoint path everywhere
e.g = 
export const PROJECT_LIST_ENDPOINT = "/api/profiles/:profile_id/projects";

use - 
PROJECT_LIST_ENDPOINT.replace(":profile_id", profile_id);

File: src/Constants.js [hunk 2]
```diff
@@ -58,7 +64,12 @@ export const SKILLS = [
 ];
 
 export const ROUTES = {
-  profile: "/api/profile/"
+  profile: "/api/profiles",
```
Comment (Payal13) [inline src/Constants.js]: what is 2 value in all routes ?

File: src/api/achievementApi.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { createApi } from "@reduxjs/toolkit/query/react";
+import { ACHIEVEMENT_ENDPOINT, PROFILE_ENDPOINT } from "../Constants";
+import axiosBaseQuery from "./axiosBaseQuery/service";
+
+export const achievementApi = createApi({
+  reducerPath: "achievementApi",
```
Comment (Payal13) [inline src/api/achievementApi.jsx]: Please add constant for all reducer paths

File: src/api/achievementApi.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  baseQuery: axiosBaseQuery(),
+  tagTypes: ["achievements"],
```
Comment (Payal13) [inline src/api/achievementApi.jsx]: add constats for tagtypes

File: src/api/certificateApi.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { createApi } from "@reduxjs/toolkit/query/react";
+import { CERTIFICATE_ENDPOINT, PROFILE_ENDPOINT } from "../Constants";
+import axiosBaseQuery from "./axiosBaseQuery/service";
+
+export const certificateApi = createApi({
+  reducerPath: "certificateApi",
```
Comment (Payal13) [inline src/api/certificateApi.jsx]: same here

File: src/api/educationApi.jsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import { EDUCATION_ENDPOINT, PROFILE_ENDPOINT } from "../Constants";
+import axiosBaseQuery from "./axiosBaseQuery/service";
+
+export const educationApi = createApi({
+  reducerPath: "educationApi",
+  baseQuery: axiosBaseQuery(),
+  tagTypes: ["educations"],
+  endpoints: (builder) => ({
+
+    getEducations :builder.query({
```
Comment (Payal13) [inline src/api/educationApi.jsx]: somethings looks wrong with placm,ent if `:`
getEducations :builder.query({

getEducations: builder.query({ ? fix everywhete

File: src/api/profileApi.jsx [hunk 7]
```diff
@@ -13,8 +13,22 @@ export const profileApi = createApi({
...
       transformResponse: (response) => response.data
+    }),
+
+    getBasicInfo: builder.query({
+      query: (id) => {
+        if (id === undefined) {
+          return { data: null };
+        }
+        return {
+          url: PROFILE_ENDPOINT + id
```
Comment (Payal13) [inline src/api/profileApi.jsx]: don't concatenate strings. use replace instead

File: src/api/projectApi.jsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+export const projectApi = createApi({
+  reducerPath: "projectApi",
+  baseQuery: axiosBaseQuery(),
+  tagTypes: ["projects"],
+  endpoints: (builder) => ({
+
+    getProject :builder.query({
+      query: (profile_id)=>({
+        url: PROFILE_ENDPOINT+profile_id+PROJECT_ENDPOINT
```
Comment (Payal13) [inline src/api/projectApi.jsx]: spacing issues. is your eslint properly configured? it should fix these issues onSave.

use replace with complete enpoint string. don't use +

File: src/api/store/store.jsx [hunk 9]
```diff
@@ -12,14 +17,19 @@ const preloadedState = {
 const rootReducer = combineReducers({
   [loginApi.reducerPath]: loginApi.reducer,
   [profileApi.reducerPath]: profileApi.reducer,
+  [projectApi.reducerPath]: projectApi.reducer,
```
Comment (Payal13) [inline src/api/store/store.jsx:20]: use constant here instead of profileApi.reducerPath

File: src/components/Builder/BasicInfo/index.jsx [hunk 10]
```diff
@@ -1,28 +1,46 @@
...
     setInitialState({
       ...initialState,
-      basicInfo: { ...initialState.basicInfo, ...values },
+      basicInfo: updatedBasicInfo,
     });
 
-    post(ROUTES.profile, initialState);
+    try {
+      await post(ROUTES.profile, { profile: updatedBasicInfo });
+      navigate(`/profiles`);
```
Comment (Payal13) [inline src/components/Builder/BasicInfo/index.jsx]: use constant here

File: src/components/Builder/Achievement/index.jsx [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+import { useParams } from "react-router-dom";
+import { Button, Form, Input, Space, Tabs } from "antd";
+import { DndContext, PointerSensor, useSensor } from "@dnd-kit/core";
+import { arrayMove, horizontalListSortingStrategy, SortableContext, useSortable } from "@dnd-kit/sortable";
+import { CSS } from "@dnd-kit/utilities";
+import { useGetAchievementsQuery } from "../../../api/achievementApi";
+import { post } from "../../../services/axios";
+import { ResumeContext } from "../../../utils/ResumeContext";
+
+const DraggableTabNode = ({ ...props }) => {
```
Comment (Payal13) [inline src/components/Builder/Achievement/index.jsx]: can we move DraggableTabNode as common component and use in all other places ?

File: src/components/Builder/Experience/index.jsx [hunk 12]
```diff
@@ -1,141 +1,279 @@
...
-            <Input placeholder="Enter Company Name Ex. Amazon" />
-          </Form.Item>
-        </Col>
-      </Row>
-      <Row style={{ margin: "10px 0px 10px 0px" }}>
-        <Col>
-          <Checkbox
-            name={"isCurrentCompany"}
-            checked={isCurrentCompany}
+    <div>
```
Comment (Payal13) [inline src/components/Builder/Experience/index.jsx]: use fragment

File: src/components/Builder/Index.jsx [hunk 13]
```diff
@@ -1,56 +1,77 @@
...
+
+  const {data} = useGetBasicInfoQuery(id);
+
+  useEffect(() => {
+    if (id) {
+      if(data){
+        setProfileData(data);
+        setItems(createPanes(data, false));
+      }
+    } else {
```
Comment (Payal13) [inline src/components/Builder/Index.jsx:71]: do we  need of else condition here ? if not remove it

File: src/components/Profile/List/index.jsx [hunk 14]
```diff
@@ -104,6 +105,10 @@ const ListProfiles = () => {
       )
   });
 
+  const handleClick = (id) => {
+    navigate(`/profile-builder/${id}`);
```
Comment (Payal13) [inline src/components/Profile/List/index.jsx]: constant please

File: src/components/Profile/List/index.jsx [hunk 15]
```diff
@@ -154,17 +160,17 @@ const ListProfiles = () => {
       dataIndex: "is_current_employee",
       key: "is_current_employee",
       render: (is_current_employee) => (
-        <strong>{is_current_employee === 1 ? "YES" : "NO"}</strong>
+        <strong>{is_current_employee}</strong>
```
Comment (Payal13) [inline src/components/Profile/List/index.jsx]: remove strong tag, instaed we can add Yes with green colour text and No with red colour text with antd component

File: src/components/Resume/jsonData.json
Comment (Payal13) [inline src/components/Resume/jsonData.json:1]: remove this file

File: src/api/profileApi.jsx [hunk 16]
```diff
@@ -1,20 +1,34 @@
...
+      transformResponse: (response) => response.data
+    }),
+
+    getBasicInfo: builder.query({
+      query: (id) => {
+        if (id === undefined) {
+          return { data: null };
+        }
+        return {
+          url: PROFILE_GET_ENDPOINT + id
```
Comment (Payal13) [inline src/api/profileApi.jsx]: do not concatinate

File: src/Constants.js [hunk 17]
```diff
@@ -117,5 +136,24 @@ export const ROOT_ROUTE = "/";
 export const PROFILE_LIST_ROUTE = "/profiles";
 export const DASHBOARD_ROUTE = "/dashboard";
 export const EDITOR_ROUTE = "/profile-builder";
-export const EDITOR_PROFILE_ROUTE = "/profile-builder/:id";
+export const EDITOR_PROFILE_ROUTE = "/profile-builder/:profile_id";
 export const ERROR_ROUTE = "*";
+
+//Draggble Tabs
+export const DraggableTabNode = ({ ...props }) => {
```
Comment (Payal13) [inline src/Constants.js]: add common components folder  and move it there

File: src/Constants.js [hunk 18]
```diff
@@ -57,13 +67,22 @@ export const SKILLS = [
...
 ];
 
-export const ROUTES = {
-  profile: "/api/profiles",
-  project:"/profiles/2/projects",
-  education:"/profiles/2/educations",
-  experience:"/profiles/2/experiences",
-  achievement:"/profiles/2/achievements",
-  certificate:"/profiles/2/certificates"
+export const ReducerPath = {
```
Comment (Payal13) [inline src/Constants.js:66]: resolve and fix

File: src/Constants.js (hunk 1 above)
Comment (Abhishek-Jain-1925) [reply src/Constants.js]: Done!

File: src/Constants.js (hunk 2 above)
Comment (Abhishek-Jain-1925) [reply src/Constants.js]: Done!

File: src/api/achievementApi.jsx (hunk 4 above)
Comment (Abhishek-Jain-1925) [reply src/api/achievementApi.jsx]: Done!

File: src/components/Builder/Experience/index.jsx (hunk 12 above)
Comment (Abhishek-Jain-1925) [reply src/components/Builder/Experience/index.jsx]: Done!

File: src/components/Builder/Index.jsx (hunk 13 above)
Comment (Abhishek-Jain-1925) [reply src/components/Builder/Index.jsx:71]: Done!

File: src/components/Profile/List/index.jsx (hunk 14 above)
Comment (Abhishek-Jain-1925) [reply src/components/Profile/List/index.jsx]: Done!

File: src/components/Profile/List/index.jsx (hunk 15 above)
Comment (Abhishek-Jain-1925) [reply src/components/Profile/List/index.jsx]: Done!

File: src/api/profileApi.jsx (hunk 16 above)
Comment (Abhishek-Jain-1925) [reply src/api/profileApi.jsx]: Done!

File: src/Constants.js (hunk 17 above)
Comment (Abhishek-Jain-1925) [reply src/Constants.js]: Done!

**Pull request merged**